tests/test_scalars.py

```diff
@@ -15,26 +22,568 @@
+    check(assert_type(as_offset + ts, pd.Timestamp), pd.Timestamp)
+
+    check(assert_type(ts + as_timedelta_index, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(as_timedelta_index + ts, pd.DatetimeIndex), pd.DatetimeIndex)
+
+    check(assert_type(ts + as_timedelta_series, "TimestampSeries"), pd.Series)
+    check(assert_type(as_timedelta_series + ts, "TimestampSeries"), pd.Series)
+
+    check(assert_type(ts + as_np_ndarray_td64, npt.NDArray[np.datetime64]), np.ndarray)
+    # pyright and mypy disagree on the type of this expression
+    # pyright: Any, mypy: npt.NDArray[np.datetime64]
+    check(
+        assert_type(
+            as_np_ndarray_td64 + ts,  # pyright: ignore [reportGeneralTypeIssues]
+            npt.NDArray[np.datetime64],
+        ),
+        np.ndarray,
+    )
+
+    # Reverse order is not possible for all of these
+    check(assert_type(ts - as_pd_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - as_dt_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - as_offset, pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - as_timedelta_index, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(ts - as_timedelta_series, "TimestampSeries"), pd.Series)
+    check(assert_type(ts - as_np_ndarray_td64, npt.NDArray[np.datetime64]), np.ndarray)
+
+
+def test_timestamp_cmp() -> None:
+    ts = pd.Timestamp(year=2000, month=3, day=24, hour=12, minute=27)
+
+    np_dt64_arr: npt.NDArray[np.datetime64] = np.array(
+        [1, 2, 3], dtype="datetime64[ns]"
+    )
+
+    c_timestamp = ts
+    c_np_dt64 = np.datetime64(1, "ns")
+    c_dt_datetime = dt.datetime(year=2000, month=1, day=1)
+    c_datetimeindex = pd.DatetimeIndex(["2000-1-1"])
+    c_np_ndarray_dt64 = np_dt64_arr
+    # Typing provided since there is no way to get a Series[Timestamp],
+    # which is a different type from a TimestampSeries
+    c_series_dt64: pd.Series[pd.Timestamp] = pd.Series(
+        [1, 2, 3], dtype="datetime64[ns]"
+    )
+    c_series_timestamp = pd.Series(pd.DatetimeIndex(["2000-1-1"]))
+    check(assert_type(c_series_timestamp, TimestampSeries), pd.Series)
+    # Use xor to ensure one is True and the other is False
+    # Correctness ensures since tested to be bools
+    gt = check(assert_type(ts > c_timestamp, bool), bool)
+    lte = check(assert_type(ts <= c_timestamp, bool), bool)
+    assert gt != lte
+
+    gt = check(assert_type(ts > c_np_dt64, bool), bool)
+    lte = check(assert_type(ts <= c_np_dt64, bool), bool)
+    assert gt != lte
+
+    gt = check(assert_type(ts > c_dt_datetime, bool), bool)
+    lte = check(assert_type(ts <= c_dt_datetime, bool), bool)
+    assert gt != lte
+
+    check(assert_type(ts > c_datetimeindex, np_ndarray_bool), np.ndarray)
+    check(assert_type(ts <= c_datetimeindex, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts > c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+    check(assert_type(ts <= c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
```

Comment:
can add either `bool` or `np.bool_` (not sure which) as additional argument to `check` in above 4 tests.

Reply:
Added